lib/widgets/message_list.dart

```diff
@@ -354,62 +376,65 @@ class MessageItem extends StatelessWidget {
-          child: MessageWithPossibleSender(item: item)),
-        if (trailing != null && item.isLastInBlock) trailing!,
-      ]));
-
-    // Web handles the left-side recipient marker in a funky way:
-    //   box-shadow: inset 3px 0px 0px -1px #c2726a, -1px 0px 0px 0px #c2726a;
-    // (where the color is the stream color.)  That is, it's a pair of
-    // box shadows.  One of them is inset.
-    //
-    // At attempt at a literal translation might look like this:
-    //
-    // DecoratedBox(
-    //   decoration: ShapeDecoration(shadows: [
-    //     BoxShadow(offset: Offset(3, 0), spreadRadius: -1, color: highlightBorderColor),
-    //     BoxShadow(offset: Offset(-1, 0), color: highlightBorderColor),
-    //   ], shape: Border.fromBorderSide(BorderSide.none)),
-    //   child: MessageWithSender(message: message)),
-    //
-    // But CSS `box-shadow` seems to not apply under the item itself, while
-    // Flutter's BoxShadow does.
+      child: _UnreadMarker(
+        isRead: message.flags.contains(MessageFlag.read),
+        child: Column(children: [
+          MessageWithPossibleSender(item: item),
+          if (trailing != null && item.isLastInBlock) trailing!,
+        ])));
+  }
+}
+
+/// Widget responsible for showing the read status of a message.
+// Mobile currently uses a single inset box-shadow:
+//   box-shadow: inset 4px 0 hsl(222, 99%, 69%);
+// The animation in mobile has a 1s delay, and a 0.3s
+// transition using the ease-out curve.
+// See zulip-mobile:src/webview/static/base.css .
+//
+// Modern web uses a block element will a linear-gradient
+// background fill:
+//   linear-gradient(90deg, hsl(217deg 64% 59%) 30%, hsl(217deg 64% 59% / 0%))
+// Transition is 0.3s for appearing and 2s for disappearing,
+// both with an ease-out curve.
+// See zulip:web/styles/message_row.css .
+//
+// Current design work is ongoing and values used here come
+// from discussion at:
+//   https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/flutter.3A.20unread.20marker
+// as well as mockups at:
+//   https://github.com/zulip/zulip-mobile/issues/5511
+// In particular, the mockups has this marker as 4px wide
+// in HSLA(227, 78%, 59%, 1). Transition speed and animation
+// curve here taken from web.
+class _UnreadMarker extends StatelessWidget {
+  const _UnreadMarker({required this.isRead, required this.child});
+
+  final bool isRead;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: [
+        child,
+        Positioned(
+          top: 0,
+          left: 0,
+          bottom: 0,
+          child: AnimatedOpacity(
+            opacity: isRead ? 0 : 1,
+            duration: Duration(milliseconds: isRead ? 2000 : 300),
+            curve: Curves.easeOut,
+            child: Container(
+              width: 4,
+              decoration: BoxDecoration(
+                color: const HSLColor.fromAHSL(1, 227, 0.78, 0.59).toColor())))),
```

Comment:
Here's a version integrating my other comment, and then putting the key pieces of information about the design into concise comments right at the relevant bits of the code:
```suggestion
          child: SizedBox(width: 4,
            child: AnimatedOpacity(
              opacity: isRead ? 0 : 1,
              // Web uses 2s and 0.3s durations, and a CSS ease-out curve.
              // See zulip:web/styles/message_row.css .
              duration: Duration(milliseconds: isRead ? 2000 : 300),
              curve: Curves.easeOut,
              child: ColoredBox(
                // The color hsl(227deg 78% 59%) comes from the Figma mockup at:
                //   https://github.com/zulip/zulip-mobile/issues/5511
                //   https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=132-9684
                color: const HSLColor.fromAHSL(1, 227, 0.78, 0.59).toColor())))),
```

Then I think that replaces most of the comment that's at the top of the class.

Reply:
Ah right or in light of the discussion above at https://github.com/zulip/zulip-flutter/pull/317#discussion_r1358910206 , the link to that GitHub issue can be replaced with a chat link:
https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/flutter.3A.20unread.20marker/near/1658008

(Best to link to a specific message ID, especially when the thread is already long. That way it's clear which part of the discussion you're referring to.)

The issue is linked from the chat thread, and it doesn't add anything specific about the colors besides the link to Figma, so the issue link can be dropped when the chat link is present.